### PR TITLE
Cosmetic changes in hclust.jl

### PR DESCRIPTION
**Pull request by davidavdav**
Just some spacing, `≤` rather than `<=`, and some double-for-loop rewrites in my own code.

### CODE REVIEW

File: test/hclust.jl [hunk 1]
```diff
@@ -1,3 +1,5 @@
+using Clustering
```
Comment (nalimilan) [inline test/hclust.jl]: If that was needed I guess the tests would fail?

**Comment by nalimilan** [comment]
+1 I prefer `in` too!

**Comment by ararslan** [comment]
Heh, I actually like `=` more than `in` but I don't care that much. 😉 I still appreciate the contribution!

File: test/hclust.jl (hunk 1 above)
Comment (davidavdav) [reply test/hclust.jl]: The test as run from `src/runtests.jl` would work all right, but this line allows a separate test
```julia
include("src/Clustering.jl")
include("test/hclust.jl")
```

Comment (ararslan) [reply test/hclust.jl]: I assume this isn't the cause of the nightly failure?
```
ERROR: LoadError: LoadError: importing Clustering into Main conflicts with an existing identifier
```

Comment (davidavdav) [reply test/hclust.jl]: All the other tests in `test/` have `using Clustering` in their code (but after `using Base.Test`---I don't assume order can matter here).  

On my machine, it is `mcl.jl` that fails on an `InexactError()`.  But that is a different issue I suppose.

Comment (ararslan) [reply test/hclust.jl]: Huh, weird. Okay.

Comment (nalimilan) [reply test/hclust.jl]: I'd check whether the error hasn't been introduced by this PR though, and if so file an issue in Base.

**Comment by alyst** [comment]
I guess the only thing that keeps the PR from merging are Travis failures.
@davidavdav Could you please rebase it to the current master to check whether the tests pass on 0.6?

**Comment by davidavdav** [comment]
I have tried, but git / hub /travis have defied me.

**Comment by alyst** [comment]
Thanks! The tests pass for 0.6 on Linux, so I'm merging.

**Pull request merged**